### ISSUE TITLE
Add TypeScript definitions for cruise result 

### ISSUE THREAD

**Issue by sverweij** [sverweij/dependency-cruiser#196]
There's no TypeScript type definition for the internal output format. Both the 
IReporterOutput interface and the format function's first parameter are `any` 
typed right now. 

This should be corrected so it has a type that matches the
results-schema.

```typescript
export interface IReporterOutput {
  /**
   * The output proper of the reporter. For most reporters this will be
   * a string. For the default reporter it is a javascript object adhering
   * to the results JSON schema in src/extract/results-schema.json
   */
  output: any; // <= this should in stead be: string | ICruiseResult

  exitCode: number;
}
```

## Context
For those using dependency-cruiser programmatically, there's type definitions for almost everything, _except_ for the actual output format - which is still typed as `any`.

### What's already done
- The output format is defined in a JSON schema: [src/extract/results-schema.json](https://github.com/sverweij/dependency-cruiser/blob/develop/src/extract/results-schema.json).
- Several of the data types that are used in the output fo (e.g. the [ICruiseOptions](https://github.com/sverweij/dependency-cruiser/blob/develop/types/dependency-cruiser.d.ts#L309) and the  [IRuleSetType](https://github.com/sverweij/dependency-cruiser/blob/develop/types/dependency-cruiser.d.ts#L256) in its meta data.

### What needs to be done?
- in [types/dependency-cruiser.d.ts](https://github.com/sverweij/dependency-cruiser/blob/develop/types/dependency-cruiser.d.ts) add an ICruiseResult interface
- use that as the type for the first parameter of the [`cruise`](https://github.com/sverweij/dependency-cruiser/blob/develop/types/dependency-cruiser.d.ts#L451) function.
- ... and use it as the type for the `output` in the [IRenderOutput](https://github.com/sverweij/dependency-cruiser/blob/develop/types/dependency-cruiser.d.ts#L409) interface (using a [union type](https://www.typescriptlang.org/docs/handbook/advanced-types.html#union-types) because that output can contain e.g. a string as well)

What should help: in the past I've had success with converting json schema to typescript definitions with [quicktype](https://github.com/quicktype/quicktype). When [tickled correctly](https://gist.github.com/sverweij/f749c01e300b705f3386533e6821c029) this also emits the comments in the json into proper typescript comments, which will helpfully show up in IDE's autocomplete.

**Comment by jayzhou3** [comment]
I am not sure what question you are trying to solve here.

**Comment by sverweij** [comment]
@jayzhou3 thanks for your interest in this issue. 

dependency-cruiser('s API) has a set of TypeScript types so it's easier to use in TypeScript, and IDE's (even non-TypeScript ones) can autocomplete. The most important one - the object format of the cruise result doesn't have these yet, though. 

It's not hard to do, especially given there's a (documented) json schema to work with. It's a good issue to start with if you're getting into writing TypeScript typings and/ or want to play with quicktype.

(I've updated the original description a bit to make this more clear - thanks for your feedback!)

**Comment by tharun208** [comment]
I want to work on this

**Comment by sverweij** [comment]
fixed in #202
